clients/unshaded/src/main/java/tachyon/client/BlockOutStream.java

```diff
@@ -61,7 +61,10 @@ public static BlockOutStream get(TachyonFile tachyonFile, WriteType opType, int
 
     if (tachyonFile.mTachyonFS.hasLocalWorker()
         && tachyonConf.getBoolean(Constants.USER_ENABLE_LOCAL_WRITE,
-            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)) {
+            Constants.DEFAULT_USER_ENABLE_LOCAL_WRITE)
+            && tachyonConf.get(
```

Comment:
Sorry, I still didn't get your reasoning. The worker is returned by master using the Balance Strategy, if the strategy is not `LocalFirstBalancer`, say, it is `MaxFreeBalancer`, and the local worker happens to be the worker with max free space, then why do we need to use `RemoteBlockOutStream` instead of `LocalBlockOutStream` in this case?